py/configu/stores/key_value_store.py

```diff
@@ -103,4 +103,7 @@ def set(self, configs: List[Config]):
             if value == "" or value is None:
                 self.delete(key)
             else:
-                self.upsert(key, json.dumps(value))
+                self.upsert(
+                    key,
+                    json.dumps(value) if isinstance(value, dict) else value,
```

Comment:
View my comment below, what happens if `value` is somehow some type other than a string and a dict? will it get properly parsed by `get`? we should make this logic airtight against unexpected edge cases.

Reply:
our guard is the schema via EvalCommand and UpsertCommand. this is why everything gets stringify.
which is missing here. thanks for that. added.